### DISCUSSION TITLE
Tensor Material Component Notation

### DISCUSSION THREAD

**Discussion by ln53** [idaholab/moose#23674] [Q&A Modules: General] [answered]
Hi,
I was using [GenericFunctionRankTwoTensor](https://mooseframework.inl.gov/moose/source/materials/GenericFunctionRankTwoTensor.html), and outputting the results to an Exodus II file. I noticed that the component indexing specified in the docs (`'fcn_00 fcn_01 fcn_02 fcn_10 fcn_11 fcn_12 fcn_20 fcn_21 fcn_22'`) were flipped when output to the Exodus file (i.e. the same components were named `'fcn_00 fcn_10 fcn_20 fcn_01 fcn_11 fcn_21 fcn_02 fcn_12 fcn_22'` in the output).

Lines 219:228 of `RankTwoTensorImplementation.h` indicate the tensor is filled in a column-major fashion (i.e. the order I experienced in the output exodus), which is opposite to the docs for GenericFunctionRankTwoTensor.

I did put together a MWE (below) to demonstrate this using [GenericConstantRankTwoTensor](https://mooseframework.inl.gov/moose/source/materials/GenericConstantRankTwoTensor.html), which shows if you input the components in a row-major order they get flipped as expected. Is this just a mistake in the docs or am I missing something?

Thanks

<details>
<summary>tensor.i</summary>

```
[Mesh]
  [mesh]
    type = GeneratedMeshGenerator
    dim = 3
  []
[]

[Variables]
  [var]
    initial_condition = 0
  []
[]

[Kernels]
  [dmdt]
    type = TimeDerivative
    variable = var
  []
[]

[Materials]
  [const]
    type = GenericConstantRankTwoTensor
    tensor_name = const
    tensor_values = '00 01 02 10 11 12 20 21 22'
    outputs = exodus
  []
[]

[Executioner]
  type = Transient
  solve_type = 'NEWTON'
  dt = 1
  end_time = 1
[]

[Outputs]
  exodus = true
[]
```

</details>

For example component 2,1 has an output value of 12:
<img width="393" alt="image" src="https://user-images.githubusercontent.com/84389483/224034714-c7505435-28c5-4ea6-bbe2-ecaaad66031e.png">

**Comment by GiudGiud** [comment]
@dschwen what is the preferred order?
This should be a simple fix

**Comment by hugary1995** [comment, marked as answer]
yes this is a documentation typo.

**Reply by GiudGiud** [reply]
ok I ll fix this. Thanks for pitching in!
